events/node_event_recorder.go

```diff
@@ -49,6 +54,16 @@ func (r *nodeEventRecorder) handleFailure(ctx context.Context, ev *event.NodeExe
 func (r *nodeEventRecorder) RecordNodeEvent(ctx context.Context, ev *event.NodeExecutionEvent, eventConfig *config.EventConfig) error {
 	var origEvent = ev
 	var rawOutputPolicy = eventConfig.RawOutputPolicy
+
+	if len(ev.GetOutputUri()) > 0 {
+		// Both outputs.pb and deck.html should be in the same folder
+		deckURI := strings.Replace(ev.GetOutputUri(), outputsFile, deckFile, 1)
+		metadata, err := r.store.Head(ctx, storage.DataReference(deckURI))
+		if err == nil && metadata.Exists() {
+			ev.DeckUri = deckURI
+		}
+	}
```

Comment:
It feels wrong to do this here... I think it should happen here: pkg/controller/nodes/task/pre_post_execution.go:118.

We should then probably do it again here: https://github.com/flyteorg/flytepropeller/blob/master/pkg/controller/nodes/task/handler.go#L525 to populate the file just like we do with output.pb...

You will need to carry this information around so that when we create the event, we don't need to check for the existence of the file again...